### DISCUSSION TITLE
Restrict ability to directly edit kata

### DISCUSSION THREAD

**Discussion by Kacarott** [codewars/codewars.com#2903] [Ideas]
This idea was prompted by a user who recently has been directly editing a bunch of kata unprompted and without any kind of discussion or review (including even changing kata names). I believe that now that we have forks which work well for suggesting and implementing changes to kata, it is time to remove the "Edit kata" button entirely.

The ability to directly edit a kata should be kept for Mods, Menders and Authors of the kata/translations, that they can fix small quick issues, however all other users would be required to publish a fork with any proposed changes.

Benefits:
- Much less people accidentally clicking "Edit kata" and forfeiting the kata
- No time spent cleaning up after "rogue" users who go around changing lots of stuff (such as the user which prompted this idea)
- Enforces the use of the safer, reviewable, and all around better process of creating forks.

**Comment by awesomeAD1** [comment]
I am Awesome and I wholeheartedly agree.

**Comment by hobovsky** [comment]
One thing to keep in mind is that there is still a couple of fields which cannot be changed with forks, and if this suggestion gets applied, they would need an intervention of a privileged user. However, I think `menders` would be perfectly able to handle such cases.

Added bonus is getting rid of a terribly unclear [privileges](https://docs.codewars.com/gamification/privileges#required-honor-points), like "Coauthor kata" or "Edit locked testcases".

**Comment by Madjosz** [comment]
Now that the contributor bug is fixed there is even less incentive to enter the editor. Also for menders since they can approve their own forks it just takes one additional click to fix via Fork compared to editor. Plus it makes the change trackable in the translations panel and not only in revisions (which by accessing also forfeits the kata).

**Comment by kazk** [comment]
The `coauthor` privilege was involved in more areas than I thought, so I made these changes for now:

- Having an unresolved issue or suggestion older than a week no longer allows a new coauthor. This should prevent a random user from editing an approved kata.
- Mods and menders can always edit the kata.

**Reply by Kacarott** [reply]
Sounds good, that should already handle the main points. What does it look like when another user attempts to click "Edit" now? If it doesn't already, perhaps it should direct them to the Fork feature.

**Reply by kazk** [reply]
They shouldn't see "Edit". I didn't change anything else, so if they open the editor directly, it should show an error.
